scripts/change_url

```diff
@@ -0,0 +1,127 @@
+
+app=$YNH_APP_INSTANCE_NAME
+
+#=================================================
+# LOAD SETTINGS
+#=================================================
+ynh_print_info "Loading installation settings..."
+
+# Needed for helper "ynh_add_nginx_config"
+final_path=$(ynh_app_setting_get $app final_path)
+
+#=================================================
+# CHECK THE SYNTAX OF THE PATHS
+#=================================================
+ynh_print_info "Check the syntax of the paths"
+
+test -n "$old_path" || old_path="/"
+test -n "$new_path" || new_path="/"
+new_path=$(ynh_normalize_url_path $new_path)
+old_path=$(ynh_normalize_url_path $old_path)
+
+#=================================================
+# CHECK WHICH PARTS SHOULD BE CHANGED
+#=================================================
+
+change_domain=0
+if [ "$old_domain" != "$new_domain" ]
+then
+	change_domain=1
+fi
+
+change_path=0
+if [ "$old_path" != "$new_path" ]
+then
+	change_path=1
+fi
+
+#=================================================
+# STANDARD MODIFICATIONS
+#=================================================
+# MODIFY URL IN NGINX CONF
+#=================================================
+ynh_print_info "Updating nginx web server configuration..."
+
+nginx_conf_path=/etc/nginx/conf.d/$old_domain.d/$app.conf
+
+# Change the path in the nginx config file
+if [ $change_path -eq 1 ]
+then
+	# Make a backup of the original nginx config file if modified
+	ynh_backup_if_checksum_is_different "$nginx_conf_path"
+	# Set global variables for nginx helper
+	domain="$old_domain"
+	path_url="$new_path"
+	# Create a dedicated nginx config
+	ynh_add_nginx_config
+fi
+
+# Change the domain for nginx
+if [ $change_domain -eq 1 ]
+then
+	# Delete file checksum for the old conf file location
+	ynh_delete_file_checksum "$nginx_conf_path"
+	mv $nginx_conf_path /etc/nginx/conf.d/$new_domain.d/$app.conf
+	# Store file checksum for the new config file location
+	ynh_store_file_checksum "/etc/nginx/conf.d/$new_domain.d/$app.conf"
+fi
+
+#=================================================
+# SPECIFIC MODIFICATIONS
+#=================================================
+# CONFIGURE
+#=================================================
+
+application_file="$final_path/app/data/_data_/_default_/configs/application.ini"
+ynh_replace_string "default_domain = .^" "default_domain = \"$new_domain\"" "$application_file"
+if [ $old_path = "/" ] && [ $new_path = "/" ]
+then
+	ynh_replace_string "$old_domain" "$new_domain" "$final_path/index.php"
+elif [ $old_path != "/" ] && [ $new_path != "/" ]
+then
+	ynh_replace_string "$old_domain$old_path" "$new_domain$new_path" "$final_path/index.php"
+elif [ $old_path != "/" ] && [ $new_path = "/" ]
+then
+	ynh_replace_string "$old_domain$old_path" "$new_domain" "$final_path/index.php"
+elif [ $old_path = "/" ] && [ $new_path != "/" ]
+then
+	ynh_replace_string "$old_domain" "$new_domain$new_path" "$final_path/index.php"
+else
+	ynh_die "Error changing the URL"
+fi
```

Comment:
Hmm something like that :
`ynh_replace_string "$old_domain${old_path%/}" "$new_domain${new_path%/}" "$final_path/index.php"`